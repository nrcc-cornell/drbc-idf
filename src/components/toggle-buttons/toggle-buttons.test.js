import React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import ToggleButtons from './toggle-buttons.component';

let currentValue = 'btn1';

const renderToggleButtons = (tab) => {
  return render(
    <ToggleButtons
      label='Test'
      currentValue={tab}
      setFunction={(newValue) => (currentValue = newValue)}
      optionsArray={[
        {
          value: 'btn1',
          ariaLabel: 'button one',
          text: 'Button 1',
        },
        {
          value: 'btn2',
          ariaLabel: 'button two',
          text: 'Button 2',
        },
      ]}
    />
  );
};

it('render matches snapshot', () => {
  expect(renderToggleButtons(currentValue)).toMatchSnapshot();
});

it('has correct button selected', async () => {
  renderToggleButtons(currentValue);
  expect(screen.getByText('Button 1').className.includes('Mui-selected')).toBe(
    true
  );
  expect(screen.getByText('Button 2').className.includes('Mui-selected')).toBe(
    false
  );
});

it('changes button on click', async () => {
  const { rerender } = renderToggleButtons(currentValue);
  await userEvent.click(screen.getByText('Button 2'));
  rerender(
    <ToggleButtons
      label='Test'
      currentValue={currentValue}
      setFunction={(newValue) => (currentValue = newValue)}
      optionsArray={[
        {
          value: 'btn1',
          ariaLabel: 'button one',
          text: 'Button 1',
        },
        {
          value: 'btn2',
          ariaLabel: 'button two',
          text: 'Button 2',
        },
      ]}
    />
  );

  expect(screen.getByText('Button 1').className.includes('Mui-selected')).toBe(
    false
  );
  expect(screen.getByText('Button 2').className.includes('Mui-selected')).toBe(
    true
  );
});
