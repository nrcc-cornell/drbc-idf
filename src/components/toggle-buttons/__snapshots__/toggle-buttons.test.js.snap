// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="toggle-buttons-container"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-4s2st-MuiTypography-root"
        >
          Test
        </p>
        <div
          aria-label="Test"
          class="MuiToggleButtonGroup-root css-1w0fyyr-MuiToggleButtonGroup-root"
          role="group"
        >
          <button
            aria-label="button one"
            aria-pressed="true"
            class="MuiButtonBase-root MuiToggleButton-root Mui-selected MuiToggleButton-sizeSmall MuiToggleButton-standard MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal css-11x2i62-MuiButtonBase-root-MuiToggleButton-root"
            tabindex="0"
            type="button"
            value="btn1"
          >
            Button 1
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            aria-label="button two"
            aria-pressed="false"
            class="MuiButtonBase-root MuiToggleButton-root MuiToggleButton-sizeSmall MuiToggleButton-standard MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal css-11x2i62-MuiButtonBase-root-MuiToggleButton-root"
            tabindex="0"
            type="button"
            value="btn2"
          >
            Button 2
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="toggle-buttons-container"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 css-4s2st-MuiTypography-root"
      >
        Test
      </p>
      <div
        aria-label="Test"
        class="MuiToggleButtonGroup-root css-1w0fyyr-MuiToggleButtonGroup-root"
        role="group"
      >
        <button
          aria-label="button one"
          aria-pressed="true"
          class="MuiButtonBase-root MuiToggleButton-root Mui-selected MuiToggleButton-sizeSmall MuiToggleButton-standard MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal css-11x2i62-MuiButtonBase-root-MuiToggleButton-root"
          tabindex="0"
          type="button"
          value="btn1"
        >
          Button 1
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <button
          aria-label="button two"
          aria-pressed="false"
          class="MuiButtonBase-root MuiToggleButton-root MuiToggleButton-sizeSmall MuiToggleButton-standard MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal css-11x2i62-MuiButtonBase-root-MuiToggleButton-root"
          tabindex="0"
          type="button"
          value="btn2"
        >
          Button 2
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
